### PR TITLE
Generic heat source and heat exchanger

### PR DESCRIPTION
**Pull request by eott-siz**

### CODE REVIEW

File: docs/resie_energy_system_components.md [hunk 1]
```diff
@@ -594,13 +594,93 @@ Modelled as [fuel boiler](resie_energy_system_components.md#fuel-boiler-fb) with
...
+ground | probes | [Geothermal Probes](resie_energy_system_components.md#geothermal-probes)
+ground | horizontal collector | [Geothermal Heat Collector](resie_energy_system_components.md#geothermal-heat-collector)
+ground | basket collector | not implemented
+ground | trensh collector | not implemented
+ground | spiral collector | not implemented
+water | groundwater well | [Generic Heat Source](resie_energy_system_components.md#generic-heat-source)
+water | surface waters | [Generic Heat Source](resie_energy_system_components.md#generic-heat-source)
+water | waste heat from industrial processes | [Generic Heat Source](resie_energy_system_components.md#generic-heat-source)
+water | waste water | [Generic Heat Source](resie_energy_system_components.md#generic-heat-source)
+water | solar thermal collector | [Generic Heat Source](resie_energy_system_components.md#generic-heat-source)
```
Comment (hsteinacker-siz) [inline docs/resie_energy_system_components.md:640]: Should we change this to a (currrently emtpy) chapter that will be written soon to keep in mind the link?

File: docs/resie_energy_system_components.md [hunk 2]
```diff
@@ -594,13 +594,93 @@ Modelled as [fuel boiler](resie_energy_system_components.md#fuel-boiler-fb) with
...
+$$\Delta T_B = T_{source,out} - T_{sink,in}$$
+
+for a counter-current heat exchanger. Similar calculations can be made for cocurrent and cross-current heat exchangers. Setting \(\Delta T_B = \alpha \cdot \Delta T_A\) allows us to disregard the unknown variables by introducing a new one in the form of \(\alpha\), which is a measure for the temperature spread across the length of the exchange area. As heat exchangers are designed for a specific range in temperatures and mass flow, \(\alpha\) is expected to change in respect to the temperatures, becoming smaller the more \(T_{source,in}\) approaches the extreme sides of the range.
+
+This leads to the following formulation:
+
+$$T_{source,in,avg} = \frac{1}{2} \cdot (T_{source,in,max} + T_{source,in,min})$$
+$$\alpha(T_{source,in}) = 1 - \frac{abs(T_{source,in,avg} - T_{source,in})}{(T_{source,in,max} - T_{source,in,min})}$$
+$$\alpha_c = min(\alpha_{max},\ \alpha(T_{source,in}))$$
+$$T_{sink,out} = T_{source,in} - L_{min} \cdot \frac{\ln{\alpha_c^{-1}}}{1-\alpha_c}$$
```
Comment (hsteinacker-siz) [inline docs/resie_energy_system_components.md:618]: What is L_{min}?

**Review by hsteinacker-siz — APPROVED** [review]

File: docs/resie_energy_system_components.md (hunk 2 above)
Comment (eott-siz) [reply docs/resie_energy_system_components.md:618]: A scaling factor that is essentially the minimum reduction in temperature when alpha is maximal (close to the average temperature). For a value of L_min = 2.0 K, at the average temperature the source input temperature is reduced by 2.11 K.

File: docs/resie_energy_system_components.md (hunk 1 above)
Comment (eott-siz) [reply docs/resie_energy_system_components.md:640]: It links to the description of the model, which comes with this PR.

**Pull request merged**